service/component-samples/reseller/src/main/java/org/fido/iot/sample/ResellerVoucherServlet.java

```diff
@@ -93,9 +93,31 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
 
     CryptoService cs = (CryptoService) getServletContext().getAttribute("cryptoservice");
     PublicKey ownerPub = cs.decode(prevOwner);
+    PublicKey nextOwner = null;
+    int keyType = prevOwner.getAsNumber(Const.PK_TYPE).intValue();
+    for (PublicKey pub : nextOwnerPublicKeys) {
+      int ownerType = cs.getPublicKeyType(pub);
+      if (ownerType == keyType) {
+        nextOwner = pub;
+        break;
+      }
+    }
+    // we didn't find an owner entry in database as per the current owner's key-type.
+    if (null == nextOwner) {
+      System.out.println("Customer entry missing for " + serialNo);
```

Comment:
instead of printlns, could we use the logger? (on line 117 too)?

Reply:
We can merge this PR now, and raise another PR to switch to logger for all println messages.